oshi-core/src/main/java/oshi/driver/windows/DeviceTree.java

```diff
@@ -145,8 +146,8 @@ public static Quintet<Set<Integer>, Map<Integer, Integer>, Map<Integer, String>,
             }
         }
         // Look for output without parents, these are top of tree
-        Set<Integer> controllerDevices = deviceIdMap.keySet().stream().filter(k -> !parentMap.containsKey(k))
-                .collect(Collectors.toSet());
+        Set<Integer> controllerDevices = deviceIdMap.keySet().stream().parallel()
+                .filter(Predicate.not(parentMap::containsKey)).collect(Collectors.toSet());
```

Comment:
This processes a filter in parallel but given it's just a hashmap lookup, probably no performance benefit (particularly for a small number of devices... 2 or 3 usually.)